hydromt/models/v1/grid.py

```diff
@@ -0,0 +1,749 @@
+"""Grid ModelComponent."""
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import xarray as xr
+from affine import Affine
+
+# TODO change import when moving this file
+from pyproj import CRS
+from shapely.geometry import box
+
+from hydromt import gis_utils, raster, workflows
+from hydromt.data_catalog import DataCatalog
+from hydromt.nodata import NoDataStrategy, _exec_nodata_strat
+from hydromt.typing import DeferedFileClose
+
+from .model_component import ModelComponent  # TODO change import when moving this file
+from .model_utils import (
+    _assert_read_mode,
+    _assert_write_mode,
+    read_nc,
+    write_nc,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class GridModelComponent(ModelComponent):
```

Comment:
GridModelComponent or GridComponent actually? It's up for discussion but I would maybe prefer the second option?

Reply:
I agree, GridComponent is a better option.